src/CoreBundle/Service/MaintenanceModeService.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace ChameleonSystem\CoreBundle\Service;
+
+use ChameleonSystem\CoreBundle\Exception\MaintenanceModeErrorException;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\DBALException;
+
+class MaintenanceModeService implements MaintenanceModeServiceInterface
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(Connection $connection)
+    {
+        $this->connection = $connection;
+    }
+
+    public function isActivated(): bool
+    {
+        return file_exists(PATH_MAINTENANCE_MODE_MARKER);
+    }
+
+    public function isActivatedInDb(): bool
+    {
+        $col = $this->connection->fetchColumn("SELECT `shutdown_websites` FROM `cms_config`");
+
+        if (false === $col) {
+            return false;
+        }
+
+        return '1' === $col['shutdown_websites'];
+    }
+
+    public function activate(): void
+    {
+        try {
+            $this->connection->executeUpdate("UPDATE `cms_config` SET `shutdown_websites` = '1'");
+        } catch (DBALException $exception) {
+            throw new MaintenanceModeErrorException('Cannot save maintenance mode flag in database', 0, $exception);
+        }
+
+        $this->addMarkerFile();
+    }
+
+    public function deactivate(): void
+    {
+        $this->removeMarkerFile();
+
+        try {
+            $this->connection->executeUpdate("UPDATE `cms_config` SET `shutdown_websites` = '0'");
+        } catch (DBALException $exception) {
+            throw new MaintenanceModeErrorException('Cannot reset maintenance mode flag in database', 0, $exception);
```

Comment:
Mini detail: Please end sentences with a dot (also in further exception messages in this class).